game.py

```diff
@@ -85,16 +132,30 @@ def check_for_collisions(player, obstacles):
     if player.y + player.height > WINDOW_HEIGHT:
         player.y = WINDOW_HEIGHT - player.height
 
-    # This needs to happen before drawing all the other objects
-    window.fill((255, 255, 255))
+    # Calculate current frame based on animation speed
+    current_time = pygame.time.get_ticks()
+    if current_time - last_frame_time >= 200:  # Change frame every 200 milliseconds
```

Comment:
Could we add a comment to this workaround. It's not very obvious why calculating the current frame based on the time works given that sometimes it'll give put the character in some random position.